assets/src/scripts/monitoring-location/components/hydrograph/selectors/discrete-data.js

```diff
@@ -51,33 +82,25 @@ const approvalCodeText = function(approvalCode) {
  *      @prop {String} parameterName
  *      @prop {String} result
  *      @prop {String} dateTime in site's time zone.
+ *      @prop {String} approvals
  */
 export const getGroundwaterLevelsTableData = createSelector(
-    getGroundwaterLevels,
-    gwLevels => {
-        if (!gwLevels) {
+    getPrimaryParameter,
+    getGroundwaterLevelPoints,
+    (parameter, gwLevels) => {
+        if (!parameter || !gwLevels) {
             return [];
         }
-        return gwLevels.values.map((point) => {
+        return gwLevels.map((point) => {
             return {
-                parameterName: gwLevels.parameter.name,
+                parameterName: parameter.name,
                 result: point.value.toString(),
                 dateTime: DateTime.fromMillis(point.dateTime, {zone: config.locationTimeZone}).toISO({
                     suppressMilliseconds: true,
                     suppressSeconds: true
                 }),
-                approvals: approvalCodeText(point.qualifiers[0])
+                approvals: point.label
             };
         });
     }
 );
-
-/*
- * Returns a selector function that returns true if any ground water
- * levels are visible.
- * @return {Function} which returns {Boolean}
- */
-export const anyVisibleGroundwaterLevels = createSelector(
```

Comment:
We don't need this anymore. Only the groundwater levels that would be rendered are fetched.

Reply:
lol. I thought this guy seemed a little superfluous.